### ISSUE TITLE
Inference [Tensorgraph]

### ISSUE THREAD

**Issue by patrickhop** [deepchem/deepchem#623]
There doesn't seem to be any relationship between labels and predictions. To the best of my understanding, the underlying implementation does reverse the normalization on the labels, so that cause is out.

```
"""
Script that trains weave models on delaney dataset.
"""
from __future__ import print_function
from __future__ import division
from __future__ import unicode_literals

import numpy as np
np.random.seed(123)
import tensorflow as tf
tf.set_random_seed(123)
import deepchem as dc

from sklearn.metrics import r2_score

# Load Delaney dataset
delaney_tasks, delaney_datasets, transformers = dc.molnet.load_delaney(
    featurizer='Weave', split='random')
train_dataset, valid_dataset, test_dataset = delaney_datasets

# Fit models
metric = dc.metrics.Metric(dc.metrics.pearson_r2_score, np.mean)

n_atom_feat = 75
n_pair_feat = 14
# Batch size of models
batch_size = 64
n_feat = 128

model = dc.models.WeaveTensorGraph(
    len(delaney_tasks),
    batch_size=batch_size,
    learning_rate=1e-3,
    use_queue=False,
    mode='regression')

model.fit(train_dataset, nb_epoch=10)
valid_scores = model.evaluate(valid_dataset, [metric], transformers)

preds = model.predict(valid_dataset, transformers)
preds = preds.reshape(-1, 1)
labels = valid_dataset.y.reshape(-1, 1)

print(preds.shape)
print(labels.shape)

print(preds[:10])
print(labels[:10])

r2 = r2_score(labels, preds)
print(r2)
```

```
Ending global_step 150: Average loss 2921.65
TIMING: model fitting took 10.252 s
computed_metrics: [0.53362490120787764]
(113, 1)
(113, 1)
[[-1.37254069]
 [-0.42701243]
 [-0.80553564]
 [-4.18219518]
 [-5.19086301]
 [-2.30783409]
 [-2.30232371]
 [-3.12602147]
 [-0.92542455]
 [-1.39340186]]
[[ 1.96615552]
 [ 0.97832998]
 [ 1.15489783]
 [-1.05935845]
 [-1.40772204]
 [-0.0619887 ]
 [ 1.62733613]
 [ 0.30546331]
 [ 0.55552156]
 [ 1.3505541 ]]
-9.77929404176
```

**Comment by patrickhop** [comment]
the last figure in the printout is the r2 value

**Comment by patrickhop** [comment]
this is resolved. i might push the fix later

**Comment by rbharath** [comment]
@patrickhop Was the fix a bug in deepchem itself or a downstream bug?

**Comment by patrickhop** [comment]
Downstream. Predict_on_batch should be used here - not predict.

Closing issue.
